docs/decisions/0004-managing-django-settings.rst

```diff
@@ -0,0 +1,97 @@
+Status
+======
+
+Accepted
+
+
+Context
+=======
+
+Settings are confusing right now and there are way too many layers. We have
+multiple python settings file that "inherit" from each other. In some cases some
+layers of the inheritance tree also pull settings from config files leading to
+more confusion.
+
+
+Decision
+========
+
+Rather than having multiple python settings files that override things
+differently we will move to just a few python files with most of the settings
+variance living in YAML config files that are managed by environment operators.
+The structure of these files try to match `OEP-45` as bast as we can. Because
+the edx-platform houses both the ``LMS`` and ``Studio`` applications, our
+implementation will differ slightly from the guidance provided by `OEP-45`.
+
+LMS and CMS settings will continue to live under ``lms/envs/...`` and ``cms/envs/...``.
+
+Underneath those folders, we will follow the layout suggested by `OEP-45`.
+
+
+To recap that here:
+
+``__init__.py`` will handle the loading of defaults, environment overrides, and ensuring that required settings are set.
+
+``defaults.py`` will house and document all default settings.
+
+``required.py`` will document and validate that required settings have been defined.
```

Comment:
Will the validation of the required settings happen within `required.py` itself or in a separate module that gets called during the Django startup phase?  My read of OEP-45 is that non-defaulted required settings are declared in `required.py`, not necessarily validated in this file.

Reply:
`The application will check that operators provided these values, and will not start unless they are set.` - This is from the description of `required.py` which I think is not very specific about where this should be but I think it makes sense to put this validation in `required.py`.  Though I think I see an issue with this that I'll ask about on the OEP-45 PR.